### PR TITLE
meta map and nf-test for Aria2

### PR DESCRIPTION
**Pull request by leoisl**
## PR checklist

Closes #XXX 

- [x] Adds meta map to Aria2 process' input and output and migrate Aria2 tests from pytest to nf-test;
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/aria2/meta.yml [hunk 1]
```diff
@@ -1,27 +1,51 @@
...
+  - http(s)
 tools:
   - "aria2":
       description: "aria2 is a lightweight multi-protocol & multi-source, cross platform download utility operated in command-line. It supports HTTP/HTTPS, FTP, SFTP, BitTorrent and Metalink."
+      homepage: "https://aria2.github.io/"
+      documentation: "https://aria2.github.io/manual/en/html/index.html"
       tool_dev_url: "https://github.com/aria2/aria2/"
-      licence: "['GPL v2']"
+      licence: ["GPL v2"]
+
```
Comment (JoseEspinosa) [inline modules/nf-core/aria2/meta.yml:16]:
```suggestion
```

File: modules/nf-core/aria2/meta.yml [hunk 2]
```diff
@@ -1,27 +1,51 @@
...
 input:
+  # Only when we have meta
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'test', single_end:false ]`
+
```
Comment (JoseEspinosa) [inline modules/nf-core/aria2/meta.yml:24]:
```suggestion
```

File: modules/nf-core/aria2/meta.yml [hunk 3]
```diff
@@ -1,27 +1,51 @@
...
   - source_url:
-      type: url
+      type: string
       description: Source URL to be downloaded
       pattern: "{http,https}*"
+
```
Comment (JoseEspinosa) [inline modules/nf-core/aria2/meta.yml:29]:
```suggestion
```

File: modules/nf-core/aria2/meta.yml [hunk 4]
```diff
@@ -1,27 +1,51 @@
...
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'test', single_end:false ]`
+
+  - downloaded_file:
+      type: file
+      description: Downloaded file from source
+      pattern: "*.*"
+
```
Comment (JoseEspinosa) [inline modules/nf-core/aria2/meta.yml:41]:
```suggestion
```

File: modules/nf-core/aria2/meta.yml [hunk 5]
```diff
@@ -1,27 +1,51 @@
...
   - versions:
       type: file
       description: File containing software versions
       pattern: "versions.yml"
-  - downloaded_file:
-      type: file
-      description: Downloaded files from source
-      pattern: "*.*"
+
```
Comment (JoseEspinosa) [inline modules/nf-core/aria2/meta.yml:46]:
```suggestion
```

File: modules/nf-core/aria2/meta.yml [hunk 6]
```diff
@@ -1,27 +1,51 @@
...
 authors:
   - "@JoseEspinosa"
```
Comment (JoseEspinosa) [inline modules/nf-core/aria2/meta.yml:48]:
```suggestion
  - "@JoseEspinosa"
  - "@leoisl"
```

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Great job! :rocket: 
Just added few small suggestions

**Comment by leoisl** [comment]
> Great job! 🚀 Just added few small suggestions

Thanks a lot, I really appreciate it! Applied them all! Please don't hesitate on making more corrections, I am very new to nextflow/nf-core and this is my first contribution!

**Review by JoseEspinosa — APPROVED** [review]
Thanks! LGTM :rocket:

**Pull request merged**